Code/GraphMol/FileParsers/FileParserUtils.h

```diff
@@ -32,26 +33,133 @@ T stripSpacesAndCast(const std::string &input, bool acceptSpaces = false) {
-                    Conformer *&conf, bool &chiralityPossible,
-                    unsigned int &nAtoms, unsigned int &nBonds,
-                    bool strictParsing = true, bool expectMEND = true);
+RDKIT_FILEPARSERS_EXPORT bool ParseV3000CTAB(
+    std::istream *inStream, unsigned int &line, RWMol *mol, Conformer *&conf,
+    bool &chiralityPossible, unsigned int &nAtoms, unsigned int &nBonds,
+    bool strictParsing = true, bool expectMEND = true);
 
 // nAtoms and nBonds are used
-RDKIT_FILEPARSERS_EXPORT bool ParseV2000CTAB(std::istream *inStream, unsigned int &line, RWMol *mol,
-                    Conformer *&conf, bool &chiralityPossible,
-                    unsigned int &nAtoms, unsigned int &nBonds,
-                    bool strictParsing = true);
+RDKIT_FILEPARSERS_EXPORT bool ParseV2000CTAB(
+    std::istream *inStream, unsigned int &line, RWMol *mol, Conformer *&conf,
+    bool &chiralityPossible, unsigned int &nAtoms, unsigned int &nBonds,
+    bool strictParsing = true);
 
 RDKIT_FILEPARSERS_EXPORT Atom *replaceAtomWithQueryAtom(RWMol *mol, Atom *atom);
+
+template <typename T>
+void applyMolListPropsToAtoms(ROMol &mol, const std::string &prefix,
+                              const std::string &missingValueMarker = "n/a") {
+  for (auto pn : mol.getPropList()) {
+    if (pn.find(prefix) == 0 && pn.size() > prefix.size()) {
+      std::string strVect = mol.getProp<std::string>(pn);
+      std::vector<std::string> tokens;
+      boost::split(tokens, strVect, boost::is_any_of(" \t\n"),
+                   boost::token_compress_on);
+      if (tokens.size() < mol.getNumAtoms()) {
+        BOOST_LOG(rdWarningLog)
+            << "Property list " << pn << " too short, only " << tokens.size()
+            << " elements found. Ignoring it." << std::endl;
+        continue;
+      }
+      std::string mv = missingValueMarker;
+      size_t first_token = 0;
+      if (tokens.size() == mol.getNumAtoms() + 1 && tokens[0].front() == '[' &&
+          tokens[0].back() == ']') {
+        mv = tokens[0].substr(1);
+        mv.pop_back();
+        first_token = 1;
+      }
+      std::string atompn = pn.substr(prefix.size());
+      for (size_t i = first_token; i < tokens.size(); ++i) {
+        if (tokens[i] != mv) {
+          T apv;
+          try {
+            apv = boost::lexical_cast<T>(tokens[i]);
+          } catch (const boost::bad_lexical_cast &) {
+            BOOST_LOG(rdWarningLog)
+                << "Value " << tokens[i] << " for property " << pn
+                << " of atom " << i << " can not be parsed. Ignoring it."
+                << std::endl;
+          }
+          unsigned int atomid = i - first_token;
+          mol.getAtomWithIdx(atomid)->setProp(atompn, apv);
+        }
+      }
+    }
+  }
+}
+inline void processMolPropertyLists(
+    ROMol &mol, const std::string &missingValueMarker = "n/a") {
+  applyMolListPropsToAtoms<std::string>(mol, "atom.prop.", missingValueMarker);
```

Comment:
I might think about moving the loop through getPropList out of the apply:

```
for(auto prop:mol.getPropList()) {
  if (prop.find("atom.prop.") == 0 ...)
     applyMolListPropsToAtoms<std::string>(mol, prop, missingValueMarker);
}
```

This would remove some work (3 extra loops and checks) and perhaps make the code a bit clearer.

Reply:
I actually think this would make the code less flexible.
It would save a bit of work, but this is not going to end up being a measurable fraction of the time required to construct a molecule, so I think that's ok